### PR TITLE
Simplify token creation.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/dotnet/roslyn/pull/72901.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@akhera99 @genlu ptal :)

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide ptal.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSyntaxGeneratorInternal.cs [hunk 1]
```diff
@@ -118,14 +120,14 @@ internal static SyntaxTokenList GetParameterModifiers(RefKind refKind, bool forF
...
+            RefKind.Ref => [RefKeyword],
             // Note: RefKind.RefReadonly == RefKind.In. Function Pointers must use the correct
             // ref kind syntax when generating for the return parameter vs other parameters.
             // The return parameter must use ref readonly, like regular methods.
-            RefKind.In when !forFunctionPointerReturnParameter => [SyntaxFactory.Token(SyntaxKind.InKeyword)],
-            RefKind.RefReadOnly when forFunctionPointerReturnParameter => [SyntaxFactory.Token(SyntaxKind.RefKeyword), SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword)],
-            RefKind.RefReadOnlyParameter => [SyntaxFactory.Token(SyntaxKind.RefKeyword), SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword)],
+            RefKind.In when !forFunctionPointerReturnParameter => [InKeyword],
+            RefKind.RefReadOnly when forFunctionPointerReturnParameter => [RefKeyword, ReadOnlyKeyword],
+            RefKind.RefReadOnlyParameter => [RefKeyword, ReadOnlyKeyword],
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSyntaxGeneratorInternal.cs:130]: it's really nice how much some code gets cleaner thans to features like collectoin expressions and these simple static tokens.

**Comment by CyrusNajmabadi** [comment]
@ToddGrun if you're interested :)

**Review by genlu — APPROVED** [review]

**Pull request merged**